mitodl/mitxonline-zendesk-theme
# Sync the mitxonline Zendesk theme with the upstream.

### What are the relevant tickets?

[#3839](https://github.com/mitodl/hq/issues/3839) 


### Description (What does it do?)
Here is the Testing URL of re-synced theme: ~~https://arbisoft7203.zendesk.com/hc/en-us~~ https://mitodlhelp.zendesk.com/hc/en-us

This PR syncs the mitxonline Zendesk theme with the upstream. It maintains the existing design sync and incorporates updates and features from the Zendesk upstream theme. To see a list of all the updates and features, please refer to the changelog [here](https://github.com/zendesk/copenhagen_theme/blob/master/CHANGELOG.md). Here are a few updates highlighted during testing:

- A clear search feature on multiple pages, retaining the feature for review, and disabled the existing search submit button on the homepage as it was conflicting with this feature. 
- A "Sign in" link for users, providing access to their profile, requests, and activities. This feature is also available on the mobile view.
  - Users can view all their requests and their statuses under "requests."
  - Users can view all their activities, such as following and contributions, such as article comments.
- Follow button for users to follow different sections such as announcement, FAQs, and etc

### How should this be manually tested?

1. Homepage Changes:

- Visit the [old mitxonline Zendesk home page](https://mitxonline.zendesk.com/hc/en-us) and compare it with the [new mitxonline Zendesk home page](https://mitodlhelp.zendesk.com/hc/en-us).
- Ensure that the clear search feature is functional on multiple pages.
- Confirm that the search submit button on the homepage is enabled and working.

2. Submit a Request Page:

- Navigate to the [old submit a request page](https://mitxonline.zendesk.com/hc/en-us/requests/new) and compare it with the [new submit a request page](https://mitodlhelp.zendesk.com/hc/en-us/requests/new).

- Ensure that the layout and functionality of the new submit a request page match the expected changes and incorporate new changes if any.

3. "Sign In" link:
- Confirm that the "Sign In" link is not visible in the test environment and will not be present when deployed to production or verify that the "Sign In" link is disabled in the header for both desktop and mobile views. 

Overall Testing:

- Conduct general testing of the Zendesk theme to ensure that all existing features are functioning correctly after the sync with the upstream.
- Test any other specific features or functionalities mentioned in the changelog or observed during the sync process.

### Screenshots (if appropriate):

**Old mitxonline zendesk home page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/f5f0ccc2-5f10-40c8-ad21-e0ef2ad2abc4)


**New mitxonline zendesk home page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/051b1eb1-b82a-4ce1-b3fc-6cd2724823a8)

**Old submit a request page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/3c0fae57-5a45-4c43-8bad-9230bb943439)

**New submit a request page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/ab619ce1-59bd-477c-b17f-756765c62f7b)

**Old mitxonline zendesk categories page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/37f78028-2a9a-4543-8ca4-7cdff7a96635)

**New mitxonline zendesk categories page:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/a4751d1d-1224-4637-83a5-77d9d0bc050b)

**Old mobile view:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/65d47c6e-261b-4d46-8024-0c7033fbd829)

**New mobile view:**
![image](https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/c9ec6eb8-712f-4b29-9e8b-3056ea5fa095)


**User profile drop down:**
<img width="1280" alt="End user profile drop down" src="https://github.com/mitodl/mitxonline-zendesk-theme/assets/41048311/24bce160-8e45-4115-ae89-ee547b6d5c09">


### Additional Context

-  While working on this, there was no workflow for auditing URLs using Lighthouse before. Now it exists but is not yet configured. Should we configure or let it go?

- For now, disabled/skipped some jobs that did not exist before and require a decision on whether to keep them disabled or not. ~~One job, "Lint commit messages," is intentionally disabled because its purpose is to check commit conventions from the base SHA to the new pull request SHA. However, in our case (of making a pull request - sync upstream with the master branch), where there are many old commits in copenhagen_theme that don't follow commit conventions before that job was introduced, this job will fail. We can re-enable it to enforce commit convention rules after the sync-upstream-master branch is merged with master.~~ After the strategy change, this will now rely on our commits made to this fork. It took another effort to follow commit linting rules(as our most of the commits don't follow them) but somehow managed the linting rules by updating commit message type. Those commit messages were fixed while resolving conflicts. We will follow the same approach mentioned above (disabling and re-enabling it to enforce commit convention rules after the PR merge) for other repositories while syncing, if that's not the case.
